server/channels/app/file_test.go

```diff
@@ -282,18 +285,53 @@ func TestCreateZipFileAndAddFiles(t *testing.T) {
 	th := Setup(t)
 	defer th.TearDown()
 
-	mockBackend := filesStoreMocks.FileBackend{}
-	mockBackend.On("WriteFile", mock.Anything, "directory-to-heaven/zip-file-name-to-heaven.zip").Return(int64(666), errors.New("only those who dare to fail greatly can ever achieve greatly"))
+	const (
+		zipName   = "zip-file-name-to-heaven.zip"
+		directory = "directory-to-heaven"
+	)
 
-	err := th.App.CreateZipFileAndAddFiles(&mockBackend, []model.FileData{}, "zip-file-name-to-heaven.zip", "directory-to-heaven")
+	t.Run("write file fails", func(t *testing.T) {
+		mockBackend := filesStoreMocks.FileBackend{}
+		mockBackend.On("WriteFile", mock.Anything, path.Join(directory, zipName)).Return(int64(666), errors.New("only those who dare to fail greatly can ever achieve greatly"))
 
-	require.Error(t, err)
-	require.Equal(t, err.Error(), "only those who dare to fail greatly can ever achieve greatly")
+		err := th.App.CreateZipFileAndAddFiles(&mockBackend, []model.FileData{}, zipName, directory)
 
-	mockBackend = filesStoreMocks.FileBackend{}
-	mockBackend.On("WriteFile", mock.Anything, "directory-to-heaven/zip-file-name-to-heaven.zip").Return(int64(666), nil)
-	err = th.App.CreateZipFileAndAddFiles(&mockBackend, []model.FileData{}, "zip-file-name-to-heaven.zip", "directory-to-heaven")
-	require.NoError(t, err)
+		require.Error(t, err)
+		require.Equal(t, err.Error(), "only those who dare to fail greatly can ever achieve greatly")
+	})
+
+	t.Run("write no file", func(t *testing.T) {
```

Comment:
question: (haven't gone any deeper), but shouldn't this fail if no files are written? also, if it is working as expected like no failing even when no files are written maybe we can also validate that the file count is zero in the zip file?

Reply:
I was also surprised that the code allows for empty archives to be created. While it seems strange to allow that case, I'm 0/5 if it's worth changing the code. We might need that case in the future.